### DISCUSSION TITLE
OpenID Connect Back-Channel Logout with next-auth

### DISCUSSION THREAD

**Discussion by purofy** [nextauthjs/next-auth#4982] [Help]
Hi, 

I have an next.js app with next-auth custom provider. 
I’m asking myself how to implement OpenID Connect Back-Channel Logout with next-auth to support the corresponding OIDC specification https://openid.net/specs/openid-connect-backchannel-1_0.html

I assume my app should offer some endpoint for this, which validates the logout-token and destroys the clients session.

I googled quite a lot how to implement this feature with next.js + next-auth, but couldn’t find any implementation examples or topics how to implement that in next.js

Could maybe somebody here help me or share some experience on that…

Thanks in advance

**Comment by martinmckenna** [comment]
@purofy did you ever figure out how to do this?

**Reply by purofy** [reply]
Hi,
Yes. In the end it was a custom endpoint for back channel logout implemented according to OIDC standard (https://openid.net/specs/openid-connect-backchannel-1_0.html)

**Reply by martinmckenna** [reply]
Can you explain a bit on how you exposed a Next.js endpoint to your authorization API?

Also are you using JWT access tokens? Did you have to blacklist the tokens somehow?

**Comment by DaveyCoulombe** [comment]
@purofy @martinmckenna  - Were you guys ever able to figure this out? I am looking into it for a feature request at work - NextJS Pages Router - NextAuth... any advice, challenges, examples would be super helpful!! Cheers!
